docs/javascript-styleguide.md

```diff
@@ -5,13 +5,35 @@
 
 [Introduction](#introduction)
 
+[Conventions](#conventions)
+
 [Airbnb's JavaScript Style Guide](#airbnbs-javascript-style-guide)
 
 [Linting](#linting)
 
 ### Introduction
 This JavaScript styleguide is provided to help you contribute to the Carbon repository in a way that other developers can understand quickly and easily. 
 
+### Conventions
+
+#### Theme Variants
+Props that changes the visual appearance of the component it **must** be called `variant`. New props **must not** be called `type` or `xyzType`.
```

Comment:
I think there's a typo, not sure the `it` should be in there

Reply:
🤦‍♂️